CONTRIBUTING.md

```diff
@@ -133,15 +136,17 @@ Differential testing should be used to compare assembly implementations with imp
 
 New features must be merged with associated tests. Bug fixes should have a corresponding test that fails without the bug fix.
 
-### Gas Metering
+### Best Practices
 
-The [Forge Gas Snapshot](https://github.com/marktoda/forge-gas-snapshot) library is used to measure the gas cost of individual actions. To ensure that the measured gas is accurate, tests have to be run using the isolate argument to generate the correct snapshot and ensure that CI passes:
+Best practices and naming conventions should be followed as outlined in the [Foundry Book](https://getfoundry.sh/forge/tests/overview).
 
-```sh
-$ forge test --isolate
-```
+### IR Compilation
+
+When the contracts are compiled via IR, tests should be compiled without IR and the contracts deployed from their bytecode to ensure quick compilation times and bytecode consistency. Check [here](https://github.com/Uniswap/foundry-template/blob/613f81c107cd2885a869dbe4afc1da4f96ed9218/foundry.toml#L22-L28) and [here](https://github.com/Uniswap/foundry-template/blob/main/test/deployers/CounterDeployer.sol) for an example.
+
+### Gas Metering
 
-When adding new functionality, a new gas snapshot should be added, preferably using `snapLastCall`.
+The [Forge Gas Snapshot](https://github.com/marktoda/forge-gas-snapshot) library is used to measure the gas cost of individual actions. Tests that measure gas should be annotated with `/// forge-config: default.isolate = true` and not be fuzzed to ensure that the gas snapshot is accurate and consistent for CI verification. All external functions should have a gas snapshot test, diverging paths within a function should have appropriate gas snapshot tests.
```

Comment:
isolate annotation in tests when gas metering to avoid having to run tests with the `--isolate` flag